src/test/java/net/snowflake/client/jdbc/ResultSetVectorLatestIT.java

```diff
@@ -0,0 +1,117 @@
+      assertEquals(vector, result);
+    }
+  }
+
+  @Test
+  public void testGetIntVectorAsLongArray() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      Long[] vector = {-1L, 5L};
+      ResultSet resultSet = stmt.executeQuery("select " + vectorToString(vector, "int"));
+      assertTrue(resultSet.next());
+      Long[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Long.class);
+      assertEquals(vector, result);
+    }
+  }
+
+  @Test
+  public void testGetFloatVectorAsFloatArray() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      Float[] vector = {-1.2f, 5.1f, 15.87f};
+      ResultSet resultSet = stmt.executeQuery("select " + vectorToString(vector, "float"));
+      assertTrue(resultSet.next());
+      Float[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Float.class);
+      assertEquals(vector, result);
+    }
+  }
+
+  @Test
+  public void testGetIntVectorFromTable() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      stmt.execute("create or replace table test_vector_int(x vector(int, 2), y int)");
+      stmt.execute("insert into test_vector_int select [3, 7]::vector(int, 2), 15");
+      ResultSet resultSet = stmt.executeQuery("select * from test_vector_int");
+      assertTrue(resultSet.next());
+      Integer[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Integer.class);
+      assertEquals(new Integer[] {3, 7}, result);
+    }
+  }
+
+  @Test
+  public void testGetFloatVectorFromTable() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      stmt.execute("create or replace table test_vector_float(x vector(float, 2), y float)");
+      stmt.execute("insert into test_vector_float select [-3, 7.1]::vector(float, 2), 20.3");
+      ResultSet resultSet = stmt.executeQuery("select * from test_vector_float");
+      assertTrue(resultSet.next());
+      Float[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Float.class);
+      assertEquals(new Float[] {-3f, 7.1f}, result);
+    }
+  }
+
+  private <T extends Number> String vectorToString(T[] vector, String vectorType) {
+    return Arrays.toString(vector) + "::vector(" + vectorType + ", " + vector.length + ")";
+  }
+
+  private void enforceQueryResultFormat(Statement stmt) throws SQLException {
+    String sql =
+        String.format(
+            "alter session set jdbc_query_result_format = '%s'", queryResultFormat.toUpperCase());
+    stmt.execute(sql);
+  }
+}
```

Comment:
I added tests for returning nulls and checking metadata.
But it is not possible to construct an empty vector since we get exception from the server: "Invalid vector dimension '0'"